app/redux/smesher/actions.ts

```diff
@@ -86,3 +87,32 @@ export const resumeSmeshing = () => async (
     ? dispatch(startSmeshing(smeshingOpts))
     : false;
 };
+
+export const updatePostProvingOpts = (
+  nonces: number,
+  threads: number
+) => async (_, getState: GetState) =>
+  eventsService.updateProvingOpts(getGenesisID(getState()), {
+    nonces,
+    threads,
+  });
+
+export const updateProvingOptsAndRestartSmeshing = (
+  nonces: number,
+  threads: number,
+  postSetupState: PostSetupState,
+  isSmeshingStarted: boolean
+) => async (dispatch: AppThDispatch) => {
+  // update node-config file
+  await dispatch(updatePostProvingOpts(nonces, threads));
+
+  // resolve an issue with Windows, when we use to much resources RAM and CPU and smeshing was stopped in config
+  // if in config smeshing ['smeshing-start'] === true
+  if (isSmeshingStarted) {
+    // if smeshing in progress postSetupState === PostSetupState.STATE_IN_PROGRESS
```

Comment:
Btw, please don't leave such useless comments +)
Better write why we need to pause smeshing only in this case.